File: Python/Best Time to Buy and Sell Stock/Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if(not prices):
            return 0
        buyPrice,profit=prices[0],0
        for i in range(1,len(prices)):
            price=prices[i]
            if(price<buyPrice):
                buyPrice=price
                continue
            if(price-buyPrice>profit):
                profit=price-buyPrice
        return profit
        

File: C#/Best Time to Buy and Sell Stock/Best Time to Buy and Sell Stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        List<int> profits=new List<int>();
        int buyPrice,sellPrice;
        for(int i=0;i<prices.Length;i++){
            buyPrice=prices[i];
            sellPrice=buyPrice;
            for(int j=i;j<prices.Length;j++){
                sellPrice=prices[j]>sellPrice?prices[j]:sellPrice;
            }
            profits.Add(sellPrice-buyPrice);
        }
        return profits.Count>0?profits.Max():0;
    }
}